PrebidMobile/PrebidServerAdapter/src/main/java/org/prebid/mobile/prebidserver/PrebidServerAdapter.java

```diff
@@ -97,16 +98,18 @@ public void onServerResponded(JSONObject response) {
                             String code = bid.getString(Settings.RESPONSE_CODE);
                             AdUnit adUnit = getAdUnitByCode(code);
                             if (adUnit != null) {
-                                String cacheId = bid.getString(Settings.RESPONSE_CACHE_ID);
                                 final double bidPrice = bid.getDouble(Settings.RESPONSE_PRICE);
                                 String bidder = bid.getString(Settings.RESPONSE_BIDDER);
                                 ArrayList<BidResponse> responseList = responses.get(adUnit);
                                 if (responseList == null) {
                                     responseList = new ArrayList<BidResponse>();
                                 }
+                                JSONObject targetingKeywords = bid.getJSONObject(Settings.RESPONSE_TARGETING);
+                                String format = targetingKeywords.getString(Settings.RESPONSE_CREATIVE);
+                                String cacheId = CacheManager.getCacheManager().saveCache(bid.toString(), format);
                                 BidResponse newBid = new BidResponse(bidPrice, cacheId);
                                 newBid.setBidderCode(bidder);
-                                JSONObject targetingKeywords = bid.getJSONObject(Settings.RESPONSE_TARGETING);
+                                newBid.addCustomKeyword(Settings.RESPONSE_CACHE_ID, cacheId);
```

Comment:
So currently we also have the `hb_cache_id_xyzbidder` keywords too. I think we need to consider that use case for deal bids. If we set the cacheId on server side still, that would eliminate the need to have to set all the keywords after we save the bid.